src/clients.rs

```diff
@@ -257,16 +273,25 @@ impl StateClient for State {
             last_height = std::cmp::max(last_height, rsp.height);
 
             let note = Note::from_slice(&rsp.note)?;
-            let ownership = vk.owns(&note);
-            let hash = ownership.then(|| note.hash());
-            let note = ownership.then_some(note);
 
-            state.cache.insert(psk, rsp.height, (note, hash))?;
+            for ssk in addresses.iter().flatten() {
+                let vk = ssk.view_key();
+                let psk = vk.public_spend_key();
+                let ownership = vk.owns(&note);
+                let nullifier = ownership.then(|| note.gen_nullifier(ssk));
+                let note = ownership.then_some(note);
+
+                state.cache.insert(psk, rsp.height, (note, nullifier))?;
```

Comment:
No need to call `.insert` here, since note is `None` and the last_height is not stored here anymore.
Removing will reduce the I/O ops, since it's not going to check if the column family exists.

Indeed, on my env with very low file descriptors availability, I'm getting the following error
```
Rocks cache database error: IO error: While open directory
```

Moving the insert inside the `if ownership` scope fix my issue

p.s: since we are moving the logic to have all the addresses at once, we can create a followup issue to create all CF needed at startup